test/integration/acme/orders_controller_test.go

```diff
@@ -41,8 +41,10 @@ import (
 )
 
 func TestAcmeOrdersController(t *testing.T) {
+	ctx, cancel := context.WithTimeout(context.Background(), time.Second*50)
+	defer cancel()
```

Comment:
Instead of using `context.Background()`, `context.TODO()` define a context per test that is bound to a timeout.
This should help to debug if there is a step in the test that takes too much time (the context will timeout instead of the whole bazel test).

Reply:
Why 50 seconds? Is that chosen to be some number of seconds less than the default test timeout?

Reply:
Haven't given much thought about the timeouts honestly. Most important is that the timeouts happen before the bazel go test timeout (so we can clearly see where the timeout error occurred) and after a normal slow run (to prevent flakiness)

Reply:
I think that 40 seconds works for most tests, so I'll maybe set all these tests to 40 seconds.